data/examples/declaration/value/function/operator-chains-out.hs

```diff
@@ -0,0 +1,245 @@
+module A (a) where
+
+-- Template
+-- x :: Int
+-- x =
+--   1 op_a
+--   2 op_a
+--   3 op_a
+--   4 op_b
+--   5 op_a
+--   6 op_a
+--   7 op_a
+--   8
+
+-- Right chain, normal case, 1 operator type
+a :: Int
+a =
+  1
+    ++ 2
+    ++ 3
+    ++ 4
+
+-- Right chain, : case, 1 operator type
+b :: Int
+b =
+  1
+    : 2
+    : 3
+    : 4
+    : 5
+    : 6
+    : 7
+    : 8
+
+-- Right chain, $ case, 1 operator type
+c :: Int
+c =
+  1 $
+    2 $
+      3 $
+        4 $
+          5 $
+            6 $
+              7 $
+                8
+
+-- Right chain, normal case, 2 operators with p(a) == p(b)
+e :: Int
+e =
+  1
+    ^^ 2
+    ^^ 3
+    ^^ 4
+    ^ 5
+    ^^ 6
+    ^^ 7
+    ^^ 8
+
+-- Right chain, normal case, 2 operators with p(a) > p(b)
+f :: Int
+f =
+  1
+    ^^ 2
+    ^^ 3
+    ^^ 4
+    ++ 5
+    ^^ 6
+    ^^ 7
+    ^^ 8
+
+-- Right chain, normal case, 2 operators with p(a) < p(b)
+g :: Int
+g =
+  1
+    ++ 2
+    ++ 3
+    ++ 4
+    ^^ 5
+    ++ 6
+    ++ 7
+    ++ 8
+
+-- Right chain, : case, 2 operators with p(:) == p(b)
+h :: Int
+h =
+  1
+    : 2
+    : 3
+    : 4
+    ++ 5
+    : 6
+    : 7
+    : 8
+
+-- Right chain, : case, 2 operators with p(:) > p(b)
+i :: Int
+i =
+  1
+    : 2
+    : 3
+    : 4
+    `seq` 5
+    : 6
+    : 7
+    : 8
+
+-- Right chain, : case, 2 operators with p(:) < p(b)
+j :: Int
+j =
+  1
+    : 2
+    : 3
+    : 4
+    ^^ 5
+    : 6
+    : 7
+    : 8
+
+-- Right chain, : case, 2 operators with p(a) == p(:)
+k :: Int
+k =
+  1
+    ++ 2
+    ++ 3
+    ++ 4
+    : 5
+    ++ 6
+    ++ 7
+    ++ 8
+
+-- Right chain, : case, 2 operators with p(a) > p(:)
+l :: Int
+l =
+  1
+    ^^ 2
+    ^^ 3
+    ^^ 4
+    : 5
+    ^^ 6
+    ^^ 7
+    ^^ 8
+
+-- Right chain, : case, 2 operators with p(a) < p(:)
+m :: Int
+m =
+  1 `seq`
```

Comment:
Why is `seq` put at the end of the line? Only `$` gets this treatment.

Reply:
I gave this treatment to any `InfixR 0` operator at the moment, so that `$`-like operators (`$!`, `$!!` ...) would be formatted in a consistent way. We could easily tweak this behaviour by changing the body of this function (and updating some comments in `p_exprOpTree`):
```haskell
-- (Operators.hs, l. 375)

-- | Indicates if an operator has InfixR 0 fixity. We special-case this class of operators because they often have, like '$', a specific 'separator' use-case, and we sometimes format them differently than other operators.
isHardSplitterOp :: FixityInfo -> Bool
```

Reply:
We agreed to extend this special case to InfixR 0 operators on monday, so I'll mark this as resolved